### PR TITLE
User can create a cookbook

### PR DESCRIPTION
**Pull request by tochman**
PT: https://www.pivotaltracker.com/story/show/169801222

* adds cookbook and favorite models
* adds favorites controller 
* request spec for adding a recipe as a favorite

### CODE REVIEW

**Review by yasmineezequiel — APPROVED** [review]

**Review by beccaburns — APPROVED** [review]

File: spec/requests/v1/user_can_favorite_a_recipe_spec.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+RSpec.describe 'POST /v1/users/{user_id}/favorite', type: :request do
+  let!(:creator) { create(:user) }
+  let(:creator_credentials) { creator.create_new_auth_token }
+  let!(:impressed_user) { create(:user) }
```
Comment (clalil) [inline spec/requests/v1/user_can_favorite_a_recipe_spec.rb:6]: Would there need to be sad path where a user is told he cannot create more than one cookbook if he tries?

**Review by shsteindorsson — APPROVED** [review]

File: spec/requests/v1/user_can_favorite_a_recipe_spec.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    end
+
+    it 'returns 201 response' do
+      expect(response).to have_http_status 201
+    end
+
+    it 'returns sucess response' do
+      expect(response_json['message']).to eq 'The recipe was successfully added to your favorites'
+    end
+  end
```
Comment (oliverochman) [inline spec/requests/v1/user_can_favorite_a_recipe_spec.rb:50]: what happens if you add a recipe to your favorites that has already added as a favorite

File: spec/requests/v1/user_can_favorite_a_recipe_spec.rb (hunk 1 above)
Comment (tochman) [reply spec/requests/v1/user_can_favorite_a_recipe_spec.rb:6]: he can not try that. The current solution checks if he has a cookbook and creates it if he hasn't.

File: spec/requests/v1/user_can_favorite_a_recipe_spec.rb (hunk 2 above)
Comment (tochman) [reply spec/requests/v1/user_can_favorite_a_recipe_spec.rb:50]: Not handled in this PR.

**Review by yasmineezequiel — APPROVED** [review]

**Pull request merged**